doc/charm++/manual.rst

```diff
@@ -2894,96 +2922,84 @@ elements migratable (see migration section above) and choose a load
-
-In *periodic load balancing mode*, a user specifies only how often load
-balancing is to occur, using +LBPeriod runtime option to specify the
-time interval.
-
-In *at sync mode*, the application invokes the load balancer explicitly
-at appropriate (generally at a pre-existing synchronization boundary) to
-trigger load balancing by inserting a function call (AtSync) in the
-application source code.
-
-In the prior two load balancing modes, users do not need to worry about
-how to start load balancing. However, in one scenario, those automatic
-load balancers will fail to work - when array elements are created by
-dynamic insertion. This is because the above two load balancing modes
-require an application to have fixed the number of objects at the time
-of load balancing. The array manager needs to maintain a head count of
-local array elements for the local barrier. In this case, the
-application must use the *manual mode* to trigger load balancer.
-
-The detailed APIs of these three methods are described as follows:
-
-#. **Periodical load balancing mode**: In the default setting, load
-   balancing happens whenever the array elements are ready, with an
-   interval of 1 second. It is desirable for the application to set a
-   larger interval using +LBPeriod runtime option. For example
-   “+LBPeriod 5.0” can be used to start load balancing roughly every 5
-   seconds. By default, array elements may be asked to migrate at any
-   time, provided that they are not in the middle of executing an entry
-   method. The array element’s variable usesAtSync being false
-   attributes to this default behavior.
-
-#. **At sync mode**: Using this method, elements can be migrated only at
-   certain points in the execution when the application invokes
-   AtSync(). In order to use the at sync mode, one should set usesAtSync
-   to true in the array element constructor. When an element is ready to
-   migrate, call AtSync()  [6]_. When all local elements call AtSync,
-   the load balancer is triggered. Once all migrations are completed,
-   the load balancer calls the virtual function
-   ArrayElement::ResumeFromSync() on each of the array elements. This
+There are two different ways to use load balancing for chare arrays to
+meet different needs of the applications. These methods are different
+in how and when a load balancing phase starts. The two methods are:
+**AtSync mode** and **periodic load balancing mode**.  In *AtSync
+mode*, the application invokes the load balancer explicitly at an
+appropriate location (generally at a pre-existing synchronization
+boundary) to trigger load balancing by inserting a function call
+(``AtSync()``) in the application source code.  In *periodic load
+balancing mode*, a user specifies only how often load balancing is to
+occur, using the *+LBPeriod* runtime option to specify the time
+interval.
+
+.. note:: In AtSync mode, Applications that use dynamic insertion or
```

Comment:
Does this mean that when using periodic load balancing we can insert freely and not have to worry about anything? It sounds like it below. Maybe this note is better inside list element `1. AtsyncMode`?

Reply:
Yes, the problem with dynamic insertion is one of of ascertaining whether all chares have called `AtSync`, so periodic completely sidesteps that. I agree that moving this into the AtSync section is better placement, I'll do that.